src/api/internal.rs

```diff
@@ -371,10 +371,22 @@ impl<T: Pixel> ContextInner<T> {
             break;
           }
 
-          self.compute_keyframe_placement(cur_lookahead_frames);
+          Self::compute_keyframe_placement(
+            cur_lookahead_frames,
+            &self.keyframes_forced,
+            &mut self.keyframe_detector,
+            &mut self.next_lookahead_frame,
+            &mut self.keyframes,
+          );
         }
       } else {
-        self.compute_keyframe_placement(&lookahead_frames);
+        Self::compute_keyframe_placement(
+          &lookahead_frames,
+          &self.keyframes_forced,
+          &mut self.keyframe_detector,
+          &mut self.next_lookahead_frame,
+          &mut self.keyframes,
```

Comment:
Did the borrow checker make you turn this into a list of parameters? 🙁

Reply:
It's been a while so I can't remember the exact error, but yeah I think it was related to the borrow checker.